sdk/communication/azure-communication-networktraversal/src/main/java/com/azure/communication/networktraversal/CommunicationRelayClientBuilder.java

```diff
@@ -0,0 +1,303 @@
+     * version based on the version of the client library being used. If no service version is specified, updating to a
+     * newer version of the client library will have the result of potentially moving to a newer service version.
+     * <p>
+     * Targeting a specific service version may also mean that the service will return an error for newer APIs.
+     *
+     * @param version {@link CommunicationRelayServiceVersion} of the service to be used when making requests.
+     * @return the updated CommunicationRelayClientBuilder object
+     */
+    public CommunicationRelayClientBuilder serviceVersion(CommunicationRelayServiceVersion version) {
+        return this;
+    }
+
+    /**
+     * Create asynchronous client applying HMACAuthenticationPolicy, UserAgentPolicy,
+     * RetryPolicy, and CookiePolicy.
+     * Additional HttpPolicies specified by additionalPolicies will be applied after them
+     *
+     * @return CommunicationRelayAsyncClient instance
+     */
+    public CommunicationRelayAsyncClient buildAsyncClient() {
+        return new CommunicationRelayAsyncClient(createServiceImpl());
+    }
+
+    /**
+     * Create synchronous client applying HmacAuthenticationPolicy, UserAgentPolicy,
+     * RetryPolicy, and CookiePolicy.
+     * Additional HttpPolicies specified by additionalPolicies will be applied after them
+     *
+     * @return CommunicationRelayClient instance
+     */
+    public CommunicationRelayClient buildClient() {
+        return new CommunicationRelayClient(createServiceImpl());
+    }
+
+    private CommunicationNetworkingClientImpl createServiceImpl() {
+        Objects.requireNonNull(endpoint);
+
+        HttpPipeline builderPipeline = this.pipeline;
+        if (this.pipeline == null) {
+            builderPipeline = createHttpPipeline(httpClient,
+                createHttpPipelineAuthPolicy(),
+                customPolicies);
+        }
+
+        CommunicationNetworkingClientImplBuilder clientBuilder = new CommunicationNetworkingClientImplBuilder();
+        clientBuilder.endpoint(endpoint)
+            .pipeline(builderPipeline);
+
+        return clientBuilder.buildClient();
+    }
+
+    private HttpPipelinePolicy createHttpPipelineAuthPolicy() {
+        if (this.tokenCredential != null && this.azureKeyCredential != null) {
+            throw logger.logExceptionAsError(
+                new IllegalArgumentException("Both 'credential' and 'accessKey' are set. Just one may be used."));
+        }
+        if (this.tokenCredential != null) {
+            return new BearerTokenAuthenticationPolicy(
+                this.tokenCredential, "https://communication.azure.com//.default");
+        } else if (this.azureKeyCredential != null) {
+            return new HmacAuthenticationPolicy(this.azureKeyCredential);
+        } else {
+            throw logger.logExceptionAsError(
+                new IllegalArgumentException("Missing credential information while building a client."));
+        }
```

Comment:
If both TokenCredential and AzureKeyCredential are set, then we should throw an IllegalStateException instead of prioritizing one form of credential over the other.

Reply:
Yeah, we are doing that on line 248 :)